terraform/appflask.py

```diff
@@ -26,6 +27,17 @@
 app.config['SECRET_KEY'] = 'cisco'
 turbo = Turbo(app)
 
+def require_api_token(func):
+    '''This function is used to block direct access to all pages unless you have a session token'''
+    @wraps(func)
+    def check_token(*args, **kwargs):
+        if 'api_session_token' not in session:
+            # If it isn't return our access denied message (you can also return a redirect or render_template)
+            return Response("Access denied")
+
+        # Otherwise just send them where they wanted to go
+        return func(*args, **kwargs)
+    return check_token
```

Comment:
The latter, is just to avoid that going to the /destroy URL destroy the cluster automatically without any user interaction. We can potentially add it after if required.